apps/desktop/desktop_native/core/src/ssh_agent/mod.rs

```diff
@@ -0,0 +1,136 @@
+use std::sync::Arc;
+
+use anyhow::Error;
+#[cfg(windows)]
+use async_stream::stream;
+#[cfg(windows)]
+use futures::stream::{Stream, StreamExt};
+
+use ssh_agent::Key;
+use std::collections::HashMap;
+use std::sync::RwLock;
+use tokio::sync::Mutex;
+
+pub mod msg;
+pub mod ssh_agent;
+pub mod russh_encoding;
+#[cfg(windows)]
+pub mod namedpipelistenerstream;
+
+#[cfg(unix)]
+use tokio::net::UnixListener;
+#[cfg(unix)]
+use homedir::my_home;
+
+static KEYSTORE: std::sync::LazyLock<ssh_agent::KeyStore> =
+    std::sync::LazyLock::new(|| ssh_agent::KeyStore(Arc::new(RwLock::new(HashMap::new()))));
+
+#[derive(Clone)]
+struct SecureAgent {
+    send_tx: tokio::sync::mpsc::Sender<String>,
+    response_rx: Arc<Mutex<tokio::sync::mpsc::Receiver<bool>>>,
+}
+
+impl ssh_agent::Agent for SecureAgent {
+    async fn confirm(&self, ssh_key: Key) -> bool {
+        // make sure we will recv our response by locking the channel
+        let mut rx_channel = self.response_rx.lock().await;
+        self.send_tx.send(ssh_key.cipher_uuid.clone()).await.unwrap();
+        let res = rx_channel.recv().await.unwrap();
+        res
+    }
+}
+
+#[cfg(unix)]
+pub async fn start_server(
+    auth_request_tx: tokio::sync::mpsc::Sender<String>,
+    auth_response_rx: Arc<Mutex<tokio::sync::mpsc::Receiver<bool>>>,
+) -> Result<(), anyhow::Error> {
+    let env_path = std::env::var("BITWARDEN_SSH_AUTH_SOCK");
+    let ssh_path = match env_path {
+        Ok(path) => path,
+        Err(_) => {
+            println!("[SSH Agent Native Module] BITWARDEN_SSH_AUTH_SOCK not set, using default path");
+            my_home()?.ok_or(Error::msg("Could not determine home directory"))?
+                .join(".bitwarden-ssh-agent.sock")
+                .to_str()
+                .ok_or(Error::msg("Could not determine home directory"))?
+                .to_string()
+        }
+    };
+    println!("[SSH Agent Native Module] Starting SSH Agent server on {:?}", ssh_path);
+    let sockname = std::path::Path::new(&ssh_path);
+    std::fs::remove_file(sockname).unwrap_or_default();
```

Comment:
I presume unwrap_or_default is used here to ignore the error if the file doesn't exist?
I think it would be clearer like this:
```suggestion
    let _ = std::fs::remove_file(sockname);
```